pkg/snapshot/copier/copier.go

```diff
@@ -138,18 +140,54 @@ func (c *Copier) copyBackups() error {
 		destSnapshotsMap[snapshot.SnapName] = snapshot
 	}
 
+	// find snapshots missing in destination
+	var snapshotsToCopy brtypes.SnapList
 	for _, snapshot := range sourceSnapshot {
-		// Copy all the snapshots (if not already copied)
 		if _, ok := destSnapshotsMap[snapshot.SnapName]; !ok {
-			c.logger.Infof("Copying %s snapshot %s...", snapshot.Kind, snapshot.SnapName)
-			if err := c.copySnapshot(snapshot); err != nil {
-				return err
-			}
+			snapshotsToCopy = append(snapshotsToCopy, snapshot)
 		} else {
 			c.logger.Infof("Skipping %s snapshot %s as it already exists", snapshot.Kind, snapshot.SnapName)
 		}
 	}
-	return nil
+
+	if len(snapshotsToCopy) == 0 {
+		return nil
+	}
+
+	// copy all missing snapshots in parallel
+	var (
+		wg        sync.WaitGroup
+		errors    = make(chan error)
+		allErrors error
+	)
+
+	for _, s := range snapshotsToCopy {
+		snapshot := s
+		wg.Add(1)
+
+		go func() {
+			defer wg.Done()
+
+			c.logger.Infof("Copying %s snapshot %s...", snapshot.Kind, snapshot.SnapName)
+			if err := c.copySnapshot(snapshot); err != nil {
+				errors <- err
+				return
+			}
+
+			c.logger.Infof("Successfully copied %s snapshot %s...", snapshot.Kind, snapshot.SnapName)
+		}()
+	}
+
+	go func() {
+		defer close(errors)
+		wg.Wait()
+	}()
+
+	for err := range errors {
+		allErrors = multierror.Append(allErrors, err)
```

Comment:
Wouldn't it be more beneficial to use https://pkg.go.dev/errors@master#Join that was introduced in `go 1.20`? It will require us to upgrade the repo to use `go 1.20`, but it helps us avoid adding an extra dependencies on `github.com/hashicorp/go-multierror` and `github.com/hashicorp/errwrap` for the same functionality.

Reply:
I could switch this over to `errors.Join`. But I would like to refrain from upgrading the repository to go `1.20` in the scope of this PR.
Alternatively, I can replace `multierror` with a plain `[]error` type, if you prefer.

Reply:
Done in a new commit